app/controllers/carts_controller.rb

```diff
@@ -68,4 +62,26 @@ def decrement_cart_quantity
     @cart.update quantity: @cart.quantity - 1
     @cart.destroy if @cart.quantity <= 0
   end
+
+  def create_cart
+    if @cart.new_record? || @cart.quantity <= 0
+      @cart = current_user.carts.new(product_id: params[:product_id],
+                                     quantity: 1)
+    else
+      @cart.quantity += 1
+    end
+  end
+
+  def init_data
+    @product_id = params[:product_id]
+    @cart ||= current_user.carts.find_or_initialize_by product_id: @product_id
```

Comment:
```suggestion
    @cart ||= current_user.carts.find_or_initialize_by product_id: params[:product_id]
```
gộp lại như thế này được không ạ?

Reply:
sau trong view lại cần dùng @product_id ấy ạ nên em tách ra để sau dùng